go-controller/pkg/cluster/master.go

```diff
@@ -14,8 +12,193 @@ import (
 	"github.com/openvswitch/ovn-kubernetes/go-controller/pkg/util"
 
 	"github.com/openshift/origin/pkg/util/netutils"
+	"github.com/sirupsen/logrus"
 )
 
+// RebuildOVNDatabase rebuilds db if HA option is enabled and OVN DB
+// doesn't exist. First It updates k8s nodes by creating a logical
+// switch for each node. Then it reads all resources from k8s and
+// creates needed resources in OVN DB. Last, it updates master node
+// ip in default namespace to trigger event on node.
+func (cluster *OvnClusterController) RebuildOVNDatabase(
+	masterNodeName string, oc *ovn.Controller) error {
+	logrus.Debugf("Rebuild OVN database for cluster nodes")
+	var err error
+	ipChange, err := cluster.checkMasterIPChange(masterNodeName)
+	if err != nil {
+		logrus.Errorf("Error when checking Master Node IP Change: %v", err)
+		return err
+	}
+
+	// If OvnHA options is enabled, make sure default namespace has the
+	// annotation of current cluster master node's overlay IP address
+	logrus.Debugf("cluster.OvnHA: %t", cluster.OvnHA)
+	if cluster.OvnHA && ipChange {
+		logrus.Debugf("HA is enabled and DB doesn't exist!")
+		// Rebuild OVN DB for the k8s nodes
+		err = cluster.UpdateDBForKubeNodes(masterNodeName)
+		if err != nil {
+			return err
+		}
+		// Rebuild OVN DB for the k8s namespaces and all the resource
+		// objects inside the namespace including pods and network
+		// policies
+		err = cluster.UpdateKubeNsObjects(oc)
+		if err != nil {
+			return err
+		}
+		// Update master node IP annotation on default namespace
+		err = cluster.UpdateMasterNodeIP(masterNodeName)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// UpdateDBForKubeNodes rebuilds ovn db for k8s nodes
+func (cluster *OvnClusterController) UpdateDBForKubeNodes(
+	masterNodeName string) error {
+	nodes, err := cluster.Kube.GetNodes()
+	if err != nil {
+		logrus.Errorf("Failed to obtain k8s nodes: %v", err)
+		return err
+	}
+	for _, node := range nodes.Items {
+		subnet, ok := node.Annotations[OvnHostSubnet]
+		if ok {
+			// Create a logical switch for the node and connect it to
```

Comment:
We are not connecting it to distributed router, right?

Reply:
You are right. Update the code, but forget to update the comments.